README.md

```diff
@@ -11,15 +11,60 @@ https://hub.docker.com/r/stshriv/kube-agent/
+e.g.
+`ci/version.sh -v <api-version> -d <build-definition> -t <PAT> -a <account> -p <project> -b <branch-name> -s <build-status> -u <username> -c <number-of-builds-to-retrieve>`
 
-## Release Definition Setup
-To setup a release definition, run setup.sh, ci.sh, and render.sh in sequential order.
+## Continuous Integration
 
-1. setup.sh installs the requisite dependencies on a shared vsts agent. If you are running on a custom agent which already has the dependencies installed you may skip calling setup.sh
+### setup.sh
+*setup.sh* is the first stage for continuous integration on a VSTS agent machine. This script installs other continuous integration dependencies on the agent machine to ensure the default VSTS agents can run the scripts. Simply add a Shell++ task to your build definition and set the script path to ci/setup.sh with no parameters.
 
-2. ci.sh handles calling get_version.py in a vsts release definition. Provide ci.sh with the same parameters as get_version.
+### version.sh
+*version.sh* handles collecting the build IDs from a specified build definition so we can tag our images using them. version.sh handles calling into get_version.py.
 
-3. render.sh is used to create a dockerfile for each version returned by get_version.py. dockerfile.template provides a baseline dockerfile for building images with available msftkube wheels.
+To run version.sh use a Shell++ task with the filepath set to ci/version.sh
+
+Parameters:
+
+| Short Command |    Full Command    |  Type  | Required |   Default Value   |           Description           |
+|---------------|--------------------|--------|----------|-------------------|---------------------------------|
+|      -a       | --account          | string | required |                   | VSTS account name               |
+|      -b       | --branch           | string |          |'refs/heads/master'| VSTS Build Branch               |
+|      -c       | --count            |  int   |          |        0          | The number of builds to return  |
+|      -d       | --definition       |  int   | required |                   | The build definition ID         |
+|      -e       | --environmentOutput| string |          |                   | Script output                   |
+|      -p       | --project          | string | required |                   | VSTS Project within the account |
+|      -s       | --buildstatus      | string |          |                   | build status filter             |
+|      -t       | --token            | string |          |     NoneType      | User token to access account    |
+|      -u       | --user             | string |          |     NoneType      | Username associated with token  |
+|      -v       | --version          | float  |          |       4.1         | VSTS REST API version           |
+|      -V       | --verbose          | boolean|          |      False        | Print verbose output            |
+
+### render.sh
```

Comment:
I've never done it but it seems like docker build can also just take a variable. 
     --build-arg list          Set build-time variables
that would be another way to do this maybe with one less script.

Reply:
I investigated this and I wanted to use that functionality, however, I couldn't get that variable as a tag for the image. I could use it to pull the different versions but accessing it outside the dockerfile in the environment that we are tagging it in (within the docker build command) wasn't doable.